### ISSUE TITLE
Question about masking

### ISSUE THREAD

**Issue by lysecret2** [rstudio/keras#210]
Hey, I am building a sequence classification model with variable lenght inputs. Now my plan was to use 0 padding and masking. On the Rstudio website it sais that "time step with the masked value will be skipped". However, this doesn't seem to be the case. Here is a small example: 

set.seed(12344)
a=c(5,4,2,0,0)

m=keras_model_sequential()

m %>% 
layer_embedding(input_dim =  6,output_dim =  2,input_length=5,mask_zero=TRUE) 

m %>% compile(
  loss = "binary_crossentropy",
  optimizer = "adam",
  metrics = "accuracy"
)

p=predict(m,t(a))

> p[1,,]
            [,1]         [,2]
[1,]  0.01595011  0.008564901
[2,] -0.04341109 -0.045552470
[3,] -0.04590027 -0.045178711
[4,]  0.02229679 -0.013458751
[5,]  0.02229679 -0.013458751

It seems to embed the timesteps 4 and 5 just like all the other timesteps. 
I tried the same with the masking layer and got the same results.

Any help is greatly appreciated!

And layers above will see the embedding at 4th and 5th time step as just another embedding and it will change the results, For example if we build an lstm layer on top.

**Comment by lysecret2** [comment]
Could it be that masking just isnt working during predict? If so it would be hard to check.

**Comment by jonathanbratt** [comment]
I'm having a similar problem. How is masking supposed to work here?
